data-analytics/dataflow-python-examples/tests/test_image_labels.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import unittest
+
+from dataflow_python_examples.image_labels import detect_labels_uri
+
+
+class TestDataGenerator(unittest.TestCase):
+    """The test cases are focused on the business logic.  In this case this is how we parse the
+    schemas, generate data and label images.
+
+    Execution Note:
+        This script is stored in professional-services/data-analytics/dataflow-python-examples/tests
+        but should be copied to professional-services/data-analytics/dataflow-python-examples/ and
+        run from there.
+    """
+    def setUp(self):
+        logging.basicConfig(level=logging.DEBUG)
+
+    def test_detect_labels_uri_return(self):
```

Comment:
This appears to be a test for string.join, rather than your own code.

Reply:
Thanks for the comment.  We struggled with what to put in this particular unit test, as the pipeline is very simple.  

We're trying to demonstrate the best practice in each pipeline of unit testing.   By design we didn't include much "business logic" in these pipelines, to keep them simple and reusable.  

I'd be reluctant to have a pipeline without unit tests, as we expect many people to copy/paste these pipelines and modify them to kickstart their development.  They're a lot more likely to follow the unit test best practice if that's what the example does.

Perhaps we add a comment here explaining that the unit tests are here as an example.  It's primarily there to demonstrate how to structure your code to separate the "business logic" from the actual pipeline building, which is better tested through integration tests.

Reply:
That's fine, there's a good reason for it being here. I just wanted to make sure it wasn't a leftover from something else.